packages/react-i18n/src/tests/i18n.test.ts

```diff
@@ -1397,6 +1397,35 @@ describe('I18n', () => {
     });
   });
 
+  describe('#formatPercentageRange()', () => {
+    it('throws an error if no minAmount is greater than maxAcount', () => {
+      const i18n = new I18n(defaultTranslations, {locale: 'en'});
+
+      expect(() => i18n.formatPercentageRange(10, 5)).toThrow(
+        'Invalid range: min (10) is greater than max (5).',
+      );
+    });
+
+    it('formats the range as a percentage if minAmount and maxAmount are equal', () => {
+      const i18n = new I18n(defaultTranslations, defaultDetails);
+      const expected = Intl.NumberFormat(defaultDetails.locale, {
+        style: 'percent',
+      }).format(5);
+      expect(i18n.formatPercentageRange(5, 5)).toBe(expected);
+    });
+
+    it('formats the range as an interval if minAmount is lower than maxAmount', () => {
+      const i18n = new I18n(defaultTranslations, defaultDetails);
+      const expected = `${Intl.NumberFormat(defaultDetails.locale).format(5)}–${Intl.NumberFormat(
+        defaultDetails.locale,
+        {
+          style: 'percent',
+        },
+      ).format(10)}`;
+      expect(i18n.formatPercentageRange(5, 10)).toBe(expected);
```

Comment:
Is `5-1,000%` really the desired output here?

(I know this assertion follows the style of other tests, but I'd much prefer it if `expected` was a hardcoded value 🤷 )

Reply:
Thanks, good catch. This is definitely not what we want. I've created a preliminary pull request to extend the `formatPercentage` method prior to continue with this work. I intend to use that in the `formatPercentageRange` method

https://github.com/Shopify/quilt/pull/2824